File: 2nd_quarter/week05/whaeun/신규 아이디 추천.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string new_id) {
    string answer = "";
    
    //1단계
    for(int i = 0; i < new_id.size(); i++){
        if(new_id[i] >= 'A' && new_id[i] <= 'Z'){
            new_id[i] += 32;
        }
    }
    
    string sub_str = "";
    
    //2단계
    for(int i = 0; i < new_id.size(); i++){
        if((new_id[i] >= 'a' && new_id[i] <= 'z') || (new_id[i] >= '0' && new_id[i] <= '9') || new_id[i] == '-' || new_id[i] == '_' || new_id[i] == '.' ){
            sub_str += new_id[i];
        }
    }
    
    answer += sub_str[0];
    
    //3 & 4단계
    for(int i = 1 ; i < sub_str.size(); i++){
        if(sub_str[i-1] == sub_str[i] && sub_str[i] == '.'){
            continue;
        }
        else{
            answer += sub_str[i];
        }
    }
    
    if(answer[0] == '.'){
        answer = answer.substr(1, answer.size() - 1);
    }
    
    if(answer[answer.size() - 1] == '.'){
        answer = answer.substr(0, answer.size() - 1);
    }
    
    //5단계
    if(answer == ""){
        answer += "a";
    }
    
    //6단계
    if(answer.size() >= 16){
        answer = answer.substr(0, 15);
    }
    
    if(answer[answer.size() - 1] == '.'){
        answer = answer.substr(0, answer.size() - 1);
    }
    
    //7단계
    if(answer.size() == 2){
        answer += answer[1];
    }
    
    if(answer.size() == 1){
        answer += answer[0];
        answer += answer[0];
    }
    
    return answer;
}

File: 2nd_quarter/week05/rash/신규아이디추천.py
import string


def solution(new_id):
    first = False
    last = False
    answer = ''

    punctuation = string.punctuation
    # 1단계
    new_id = new_id.lower()

    # 2단계
    symbols = punctuation.replace("-", "").replace("_", "").replace(".", "")
    for symbol in symbols:
        new_id = new_id.replace(symbol, "")

    # 3단계
    while ".." in new_id:
        new_id = new_id.replace('..', '.')

    # 4단계
    if len(new_id) > 0:
        if new_id[0] == ".":
            new_id = new_id[1:]
    if len(new_id) > 0:
        if new_id[-1] == ".":
            new_id = new_id[:-1]

    # 5단계
    if new_id == "":
        new_id = "a"

    # 6단계
    if len(new_id) > 15:
        new_id = new_id[0:15]

    # 7단계
    while (len(new_id) < 3):
        new_id += new_id[-1]

    if len(new_id) > 0:
        if new_id[0] == ".":
            new_id = new_id[1:]
    if len(new_id) > 0:
        if new_id[-1] == ".":
            new_id = new_id[:-1]

    answer = new_id
    return answer